src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicPreferStreamAsyncMemoryOverloads.Fixer.vb

```diff
@@ -59,5 +59,45 @@ Namespace Microsoft.NetCore.VisualBasic.Analyzers.Runtime
 
         End Function
 
+        Protected Overrides Function IsPassingZeroAndBufferLength(model As SemanticModel, bufferValueNode As SyntaxNode, offsetValueNode As SyntaxNode, countValueNode As SyntaxNode) As Boolean
+
+            ' First argument should be an identifier name node
+            Dim firstArgumentIdentifierName = TryCast(bufferValueNode, IdentifierNameSyntax)
+            If firstArgumentIdentifierName Is Nothing Then
+                Return False
+            End If
+
+            ' Second argument should be a literal expression node with a constant value...
+            Dim optionalValue As [Optional](Of Object) = model.GetConstantValue(offsetValueNode)
+
+            ' And must be an integer...
+            If Not optionalValue.HasValue Or TypeOf optionalValue.Value IsNot Integer Then
```

Comment:
@Youssef1313 I'll also test your suggestion here (`optionalValue.Value IsNot 0`), it should probably work on VB too.